floating_window_snapping.py

```diff
@@ -0,0 +1,74 @@
+"""
+Qtile plugin that allows a moving floating window's borders
+to snap to other windows' borders and screen borders.
+
+Usage:
+1. Put this file in the same folder as the Qtile config.py file
+usually ~/.config/qtile/
+2. In config put:
+from floating_window_snapping import move_snap_window
+...
+mouse = [
+    Drag([mod], "Button1", move_snap_window(snap_dist=20),
+        start=lazy.window.get_position()),
+]
+3. snap_dist (default 20) controls how many pixels the mouse has to move
+before a snapped window will un-snap.
+~10 for light strength snapping, ~20-30 for medium and higher for high res
+displays or high dpi mice.
+
+Tested with Qtile versions 17-19
+TODO: border snapping on floating window resize
+"""
+
+from libqtile.lazy import lazy
+
+
+@lazy.window.function
+def move_snap_window(window, x, y, snap_dist=20):
+    """Move floating window to x and y.
+    Border snapping makes floating window's borders
+    stick to other borders for easy alignment
+    """
+    def _get_borders(window):
+        """Generate list of 4-tuples describing
+        the borders of every window and screen.
+        """
+        borders = []
+        for s in window.qtile.screens:
+            borders.append((s.x, s.y, s.x + s.width, s.y + s.height))
+            for w in s.group.windows:
+                if not w.hidden:
+                    borders.append(w.edges)
+        borders.remove(window.edges)
+        return borders
+
+    def _borders_touch(window, x, y, snap_dist):
```

Comment:
Looks good! Could I suggest moving this function and `_get_borders` to the top level of the file? Otherwise they are getting re-defined every time `move_snap_window` is called

Reply:
good catch, thanks :)